src/common/src/collection/evictable.rs

```diff
@@ -13,24 +13,31 @@
 // limitations under the License.
 
 use std::cmp::Eq;
-use std::hash::Hash;
+use std::hash::{BuildHasher, Hash};
 use std::ops::{Deref, DerefMut};
 
-use lru::LruCache;
+use lru::{DefaultHasher, LruCache};
 
 /// A wrapper for [`LruCache`] which provides manual eviction.
-pub struct EvictableHashMap<K, V> {
-    inner: LruCache<K, V>,
+pub struct EvictableHashMap<K, V, S = DefaultHasher> {
+    inner: LruCache<K, V, S>,
 
     /// Target capacity to keep when calling `evict_to_target_cap`.
     target_cap: usize,
 }
 
 impl<K: Hash + Eq, V> EvictableHashMap<K, V> {
     /// Create a [`EvictableHashMap`] with the given target capacity.
-    pub fn new(target_cap: usize) -> Self {
+    pub fn new(target_cap: usize) -> EvictableHashMap<K, V> {
+        EvictableHashMap::new_with_hasher(target_cap, DefaultHasher::new())
+    }
+}
+
+impl<K: Hash + Eq, V, S: BuildHasher> EvictableHashMap<K, V, S> {
+    /// Create a [`EvictableHashMap`] with the given target capacity and haser.
+    pub fn new_with_hasher(target_cap: usize, hasher: S) -> Self {
```

Comment:
`with_hasher` is recommended, see https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430]

Reply:
Also we have other new_with_xxx in our repo. May fix those later.